integrations/kubecost/client.py

```diff
@@ -39,14 +43,9 @@ async def get_kubesystem_cost_allocation(self) -> list[dict[str, Any]]:
         """Calls the Kubecost allocation endpoint to return data for cost and usage
         https://docs.kubecost.com/apis/apis-overview/api-allocation
         """
-        if self.kubecost_api_version == KUBECOST_API_VERSION_1:
-            selector = typing.cast(
-                KubecostV1ResourceConfig, event.resource_config
-            ).selector
-        else:
-            selector = typing.cast(
-                KubecostV2ResourceConfig, event.resource_config
-            ).selector
+        selector = typing.cast(
+            KubecostV1ResourceConfig | KubecostV2ResourceConfig, event.resource_config
+        ).selector
```

Comment:
this shouldn't be part of the client, but rather on the resync side

Reply:
How do I do that?

Reply:
you can create a api_utils.py that will have a method called `transform_kubecost_resource_mapping_to_api_params`

Reply:
So, you can't pass the types as an argument, the linter will complain. I put the selector code in the resync side. I'll push so you can see it

Reply:
not sure I understand what you've tried performing that it failed on linting

Reply:
can you show an example?